src/main/java/org/apache/commons/imaging/formats/webp/WebPImageParser.java

```diff
@@ -0,0 +1,351 @@
+            throw new IOException("Not a Valid WebP File");
+        }
+
+        int fileSize = read4Bytes("File Size", is, "Not a Valid WebP File", ByteOrder.LITTLE_ENDIAN);
+        if (fileSize < 0) {
+            throw new ImageReadException("File Size is too long:" + fileSize);
+        }
+
+        if (is.read(buffer) < 4 || !WebPConstants.WEBP_SIGNATURE.equals(buffer)) {
+            throw new IOException("Not a Valid WebP File");
+        }
+
+        return fileSize;
+    }
+
+    @Override
+    public WebPImageMetadata getMetadata(ByteSource byteSource, WebPImagingParameters params) throws ImageReadException, IOException {
+        try (ChunksReader reader = new ChunksReader(byteSource, WebPChunkXMP.TYPE_EXIF)) {
+            WebPChunk chunk = reader.readChunk();
+            return chunk == null ? null : new WebPImageMetadata((TiffImageMetadata) new TiffImageParser().getMetadata(chunk.getBytes()));
+        }
+    }
+
+    @Override
+    public String getXmpXml(ByteSource byteSource, XmpImagingParameters params) throws ImageReadException, IOException {
+        try (ChunksReader reader = new ChunksReader(byteSource, WebPChunkXMP.TYPE_XMP)) {
+            WebPChunkXMP chunk = (WebPChunkXMP) reader.readChunk();
+            return chunk == null ? null : chunk.getXml();
+        }
+    }
+
+    @Override
+    public ImageInfo getImageInfo(ByteSource byteSource, WebPImagingParameters params) throws ImageReadException, IOException {
+        try (ChunksReader reader = new ChunksReader(byteSource,
+                WebPChunk.TYPE_VP8, WebPChunk.TYPE_VP8L, WebPChunk.TYPE_VP8X, WebPChunk.TYPE_ANMF)) {
+            String formatDetails;
+            int width;
+            int height;
+            int numberOfImages;
+            boolean hasAlpha = false;
+            ImageInfo.ColorType colorType = ImageInfo.ColorType.RGB;
+
+            WebPChunk chunk = reader.readChunk();
+            if (chunk instanceof WebPChunkVP8) {
+                formatDetails = "WebP/Lossy";
+                numberOfImages = 1;
+
+                WebPChunkVP8 vp8 = (WebPChunkVP8) chunk;
+                width = vp8.getWidth();
+                height = vp8.getHeight();
+                colorType = ImageInfo.ColorType.YCbCr;
+            } else if (chunk instanceof WebPChunkVP8L) {
+                formatDetails = "WebP/Lossless";
+                numberOfImages = 1;
+
+                WebPChunkVP8L vp8l = (WebPChunkVP8L) chunk;
+                width = vp8l.getImageWidth();
+                height = vp8l.getImageHeight();
+            } else if (chunk instanceof WebPChunkVP8X) {
+                WebPChunkVP8X vp8x = (WebPChunkVP8X) chunk;
+                width = vp8x.getCanvasWidth();
+                height = vp8x.getCanvasHeight();
+                hasAlpha = ((WebPChunkVP8X) chunk).isHasAlpha();
+
+                if (vp8x.isAnimation()) {
+                    formatDetails = "WebP/Animation";
```

Comment:
Worth moving these strings to constants?

Reply:
> Worth moving these strings to constants?

I checked the implementation of other image formats. Format details seems to be an implementation specific internal representation, so I don't think it needs to be placed elsewhere.

Reply:
I think these could be constants in the class, like the extensions. Will move them as it's harmless :man_shrugging:

Reply:
Oh, we could even use an enum for the formats... but now I see they are being used once, and the performance benefit would be minimal. We can worry about that later. Leaving as-is, resolving this comment.